VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+                return True
+            else:
+                self.logger.log('failed to remove ' + dm_name)
+        else:
+            self.logger.log('no resource disk device mapper found')
+        return False
+
+    def prepare_partition(self):
+        """ create partition on resource disk if missing """
+        if self.resource_disk_partition_exists():
+            return True
+        self.logger.log("resource disk partition does not exist", level='Info')
+        cmd = 'parted ' + self.RD_BASE_DEV_PATH + ' mkpart primary ext4 0% 100%'
+        if self.executor.ExecuteInBash(cmd) == CommonVariables.process_success:
+            # wait for the corresponding udev name to become available
+            for i in range(0, 10):
+                time.sleep(i)
+                if self.resource_disk_partition_exists():
+                    return True
+        self.logger.log('unable to make resource disk partition')
+        return False
+
+
+    def clear_luks_header(self):
+        """ clear luks header by overwriting with 10MB of entropy """
+        if not self.resource_disk_partition_exists():
+            self.logger.log("resource partition does not exist, no luks header to clear")
+            return True
+        cmd = 'dd if=/dev/urandom of=' + self.RD_DEV_PATH + ' bs=512 count=20480'
+        return self.executor.Execute(cmd) == CommonVariables.process_success
+            
+    def try_remount(self):
+        """ mount encrypted resource disk if not already mounted"""
+        if self.is_crypt_mounted():
+            self.logger.log("resource disk already encrypted and mounted", level='Info')
+            return True
+
+        if self.resource_disk_exists() and self.resource_disk_partition_exists() and self.is_luks_device() and self.is_valid_key():
+            # store the currently associated path and name
+            current_mapper_name = self.get_rd_device_mapper()
+            if current_mapper_name:
+                self.mapper_name = current_mapper_name
+                self.mapper_path = self.DM_PREFIX + self.mapper_name
+                if not self.is_luks_device_opened:
+                    # attempt to open
+                    self.disk_util.luks_open(passphrase_file=self.RD_KEY_FILE, dev_path=self.RD_DEV_PATH, mapper_name=self.mapper_name, header_file=None, uses_cleartext_key=False)
+                    if not self.is_luks_device_opened:
+                        return False
+                # attempt mount
+                return self.mount()
+
+        # conditions required to re-mount were not met
+        return False
+
+    def prepare(self):
+        """ prepare a non-encrypted resource disk to be encrypted """
+        self.configure_waagent()
+        self.configure_fstab()
+        if self.resource_disk_partition_exists():
+            self.disk_util.swapoff()
+            self.unmount_resource_disk()
+            self.remove_device_mapper()
+            self.clear_luks_header()
```

Comment:
why do we clear the luks header again? Won't the luksFormat immediately afterwards take care of this?
Even if we do have to do it- why use urandom instead of /dev/zero

Reply:
Wiping the header is done to prevent luksFormat from entering any logic that deals with choices of what to do with a pre-existing header.  For the wipe,  /dev/zero would also work, urandom was chosen as per the following example:  https://wiki.archlinux.org/index.php/Dm-crypt/Drive_preparation#Wipe_LUKS_header